File: 74/C.py
# encoding:utf-8
import copy
import numpy as np
import random
import bisect #bisect_left　これで二部探索の大小検索が行える
import math

A,B,C,D,E,F = map(int,input().split())

water = []
for ai in range(int(F/(A*100))+1):
    for bi in range(int(F/(B*100))+1):
        if (ai*A+bi*B)*100>F:pass
        else:
            water.append((ai*A+bi*B)*100)

suger = []
for ci in range(int(F/C)+1):
    for di in range(int(F/D)+1):
        if ci*C+di*D > F:
            pass
        else:suger.append(ci*C+di*D)

water = list(set(water))
suger = list(set(suger))
water.remove(0)
pure = 0
ans_s,ans_w = 0,A*100
for wi in range(len(water)):
    for si in range(len(suger)):
        if water[wi]+suger[si]>F:pass
        else:
            if pure<suger[si]/(water[wi]+suger[si])<=(E/(100+E)):
                ans_s = suger[si]
                ans_w = water[wi]
                pure = suger[si]/(water[wi]+suger[si])

print(ans_s+ans_w,end="")
print(" ",end="")
print(ans_s)

File: 74/C.cpp
//encoding:utf-8
//#include <iostream> //c++
//#include<stack> //スタックに関して　最新のデータから捨てる
//#include<queue> //キューに関して　最古のデータから捨てる
//#include<math.h>
//#include<vector>
//#include<algorithm> //sort とかにいる
#include<bits/stdc++.h>//vector,algorithm,que
#include<time.h> //時間にかんして

#define ll long long
#define MOD 1000000007
using namespace std;
int main(){
  int A,B,C,D,E,F;
  cin>>A>>B>>C>>D>>E>>F;
  vector<int> water;
  for(int a=1;a<(int)F/(A*100);a++){
    for(int b=0;b<(int)F/(B*100);b++){
      water.push_back((a*A+b*B)*100);
    }
  }



  vector<int> suger;
  for(int c=0;c<(int)F/C;c++){
    for(int d=0;d<(int)(F-c)/D;d++){
      suger.push_back(c*C+d*D);
    }
  }

  long pure = 0;
  int ans_w=0,ans_s=0;
  for(int wi=0;wi<water.size();wi++){
    for(int si=0;si<suger.size();si++){
      if(water[wi]+suger[si]>F)break;
      else{
        cout<<"hoge"<<(suger[si]/(water[wi]+suger[si]))<<endl;
        if(pure<(suger[si]/(water[wi]+suger[si]))<(E/(100+E))){
          ans_w = water[wi];
          ans_s = suger[si];
          pure = suger[si]/(water[wi]+suger[si]);
        }
      }
    }
  }

  cout<<ans_w<<ans_s<<endl;
  return 0;
}
